examples/access_traits/main.cpp

```diff
@@ -0,0 +1,82 @@
+#include <ArborX.hpp>
+
+#include <Kokkos_Core.hpp>
+
+struct Data
+{
+  float *d_x;
+  float *d_y;
+  float *d_z;
+  int N;
+};
+
+struct Data2
+{
+  float *d_x;
+  float *d_y;
+  float *d_z;
+  float *d_r;
+  int N;
+};
+
+namespace ArborX
+{
+namespace Details
+{
+namespace Traits
+{
+template <>
+struct Access<Data>
+{
+  __device__ inline static Point get(Data const &d, int i)
+  {
+    return {{d.d_x[i], d.d_y[i], d.d_z[i]}};
+  }
+  static typename std::size_t size(Data const &d) { return d.N; }
+  using Tag = PointTag;
+  using MemorySpace = Kokkos::CudaSpace;
+};
+template <>
+struct Access<Data2>
+{
+  __device__ inline static Box get(Data2 const &d, int i)
+  {
+    Point const p{{d.d_x[i], d.d_y[i], d.d_z[i]}};
+    float const r = d.d_r[i];
+    return {{{p[0] - r, p[1] - r, p[2] - r}}, {{p[0] + r, p[1] + r, p[2] + r}}};
+  }
+  static typename std::size_t size(Data2 const &d) { return d.N; }
+  using Tag = BoxTag;
+  using MemorySpace = Kokkos::CudaSpace;
+};
+} // namespace Traits
+} // namespace Details
+} // namespace ArborX
+
+int main(int argc, char *argv[])
+{
+  Kokkos::initialize(argc, argv);
+
+  int const N = 10000;
+  float *a;
+  float *d_a;
+
+  a = (float *)malloc(sizeof(float) * N);
+  cudaMalloc((void **)&d_a, sizeof(float) * N);
```

Comment:
Why the cast? Never seen anyone doing that.

Reply:
This is how NVIDIA documents calling `cudaMalloc`